internal/datastore/flair_template.go

```diff
@@ -48,27 +48,38 @@ func (d *db) GetFlairTemplateByID(ctx context.Context, id string) (*model.FlairT
 }
 
 func (d *db) RemoveFlairTemplate(ctx context.Context, flairTemplate model.FlairTemplate) (*model.FlairTemplate, error) {
-    logrus.Debug("RemoveFlairTemplate::SQL Query")
-    // Ensure that flairTemplate.ID is set, otherwise GORM could delete all flairTemplate
-    if &flairTemplate.ID == nil {
-        logrus.Errorf("Attempted to delete flair template with no ID")
-        return &flairTemplate, nil
-    }
-    err := d.sql.Transaction(func(tx *gorm.DB) error {
-    	// Delete all the flairs using this template
-        if err := tx.Where(model.Flair{TemplateID: flairTemplate.ID}).
-        			 Delete([]model.Flair{}).Error; err != nil {
-	        return  err
-	    }
-    	// Delete the template
-        if err := tx.Delete(&flairTemplate).Error; err != nil {
-            return err
-        }
-        return nil
-    })
-    if err != nil {
+	logrus.Debug("RemoveFlairTemplate::SQL Query")
+	// Ensure that flairTemplate.ID is set, otherwise GORM could delete all flairTemplate
+	if &flairTemplate.ID == nil {
+		logrus.Errorf("Attempted to delete flair template with no ID")
+		return &flairTemplate, nil
+	}
+	err := d.sql.Transaction(func(tx *gorm.DB) error {
+		// Set Null all participants referencing the flairs we just deleted.
+		if err := tx.Unscoped().Model(model.Participant{}).
+					 Joins("FROM flairs").
+					 Where("flair_id = flairs.id AND flairs.template_id = ?", flairTemplate.ID).
+					 Update("flair_id", nil).Error; err != nil {
```

Comment:
This is the important piece of logic added in this commit. It, in addition to deleting the flairs and flair template, now sets the participant.flair_id to NULL.

Here's the sql that's created when you run removeFlairTemplate
```sql
UPDATE "participants" SET "flair_id" = NULL, "updated_at" = '2020-05-01 13:34:00'
FROM flairs
WHERE (flair_id = flairs.id AND flairs.template_id = '275284fa-a8b5-4634-b2c9-a52a8b9bdb42');

UPDATE "flairs" SET "deleted_at"='2020-05-01 13:34:00' 
WHERE "flairs"."deleted_at" IS NULL AND (("flairs"."template_id" = '275284fa-a8b5-4634-b2c9-a52a8b9bdb42'));

UPDATE "flair_templates" SET "deleted_at"='2020-05-01 13:34:00' 
WHERE "flair_templates"."deleted_at" IS NULL AND "flair_templates"."id" = '275284fa-a8b5-4634-b2c9-a52a8b9bdb42';
```

Reply:
👍